client/text.cpp

```diff
@@ -1400,128 +1401,256 @@ QString text_happiness_wonders(const struct city *pcity)
-      str += QString(PL_("%1 content before penalty.",
-                         "%1 content before penalty.", content))
-                 .arg(QString::number(content))
-             + qendl();
-      str += QString(PL_("%1 additional unhappy citizen.",
-                         "%1 additional unhappy citizens.", unhappy))
-                 .arg(QString::number(unhappy))
-             + qendl();
-      if (angry > 0) {
-        str += QString(PL_("%1 angry citizen.", "%1 angry citizens.", angry))
-                   .arg(QString::number(angry))
-               + qendl();
+        str += QString(PL_("Afterwards, for every %1 additional city, a "
+                           "content citizen becomes unhappy.",
+                           "Afterwards, for every %1 additional cities, a "
+                           "content citizen becomes unhappy.",
+                           step))
+                   .arg(step);
+
+        if (game.info.angrycitizen) {
+          str += QStringLiteral(" ");
+          str += _("If there are no more content citizens, an unhappy "
+                   "citizen becomes angry instead.");
+        }
       }
-    } else {
-      // TRANS: appended to "Cities: %d total:"; preserve leading space.
-      str +=
-          QString(PL_(" not more than %1, so no empire size penalty.",
-                      " not more than %1, so no empire size penalty.", next))
-              .arg(QString::number(next));
-      str += QString(PL_("%1 content per city.", "%1 content per city.",
-                         content))
-                 .arg(QString::number(content))
-             + qendl();
     }
-    if (next >= cities && penalty < content) {
-      str += QString(PL_("With %1 more city, another citizen will become "
-                         "unhappy.",
-                         "With %1 more cities, another citizen will become "
+  }
+  return str;
+}
+
+/**
+ * Describes what the empire size rules imply for an empire of a given size.
+ */
+QString text_happiness_cities_apply_rules(int cities, int max_content)
+{
+  if (max_content > 0) {
+    // TRANS: Pluralized in "%2 content citizens"
+    return QString(PL_("You have %1 cities, resulting in a maximum of %2 "
+                       "content citizen.",
+                       "You have %1 cities, resulting in a maximum of %2 "
+                       "content citizens.",
+                       max_content))
+        .arg(cities)
+        .arg(max_content);
+  } else if (max_content == 0) {
+    return _("You have %1 cities, thus all citizens are unhappy.");
+  } else {
+    // TRANS: Pluralized in "%2 angry citizens"
+    return QString(PL_("You have %1 cities, resulting in a maximum of "
+                       "%2 angry citizen.",
+                       "You have %1 cities, resulting in a maximum of "
+                       "%2 angry citizens.",
+                       max_content))
+        .arg(cities)
+        .arg(-max_content);
+  }
+}
```

Comment:
Does this account for a player having only one city? When that happens, it should be "You have %1 city, ..."

Reply:
Unfortunately I don't know of any library allowing pluralization twice in the same sentence :(. Maybe we can rephrase it to avoid "%1 cities" altogether?

Reply:
It could be as two separate sentences, so you pluralize each of them separately

Reply:
Unfortunately not. Usual pluralization issue :/

Reply:
Proper fix not possible before #1212